qiskit_ibm_runtime/base_primitive.py

```diff
@@ -98,6 +101,13 @@ def __init__(
         if isinstance(backend, IBMBackend):
             self._service = backend.service
             self._backend = backend
+        elif isinstance(backend, AerSimulator):
+            self._backend = backend
+            self._service = (
+                QiskitRuntimeService()
+                if QiskitRuntimeService.global_service is None
+                else QiskitRuntimeService.global_service
+            )
```

Comment:
Instead of adding all the local logic into `QiskitRuntimeService`, it'd be much cleaner to just create a new `LocalRuntimeService` to "mock" any server requests.

Reply:
I am not sure I understand your intention in this comment. Do you mean that when running on a simulator, we don't need a real service, and don't need to have a real account?

Reply:
Yes, so instead of modifying `QiskitRuntimeService`, `base_primitive` can set `self._service` to a mock (local) service. And its `run()` method would handle the local operation.

Reply:
I will try this. Are you also saying that the local run will not need an account? because in a comment above, @kt474 thought it should have an account.

Reply:
We can do it this way, without an account

Reply:
I didn't get a chance to complete this item. I think it might be best to do in a separate PR. In any case, it is up to you @kt474 or @jyu00 to continue from here.